SROS2_Linux.md

```diff
@@ -0,0 +1,201 @@
+# Try SROS2 in Linux
+
+## Installation
+
+### Install from debian packages
+
+First install ROS2 from binaries following [these instructions](https://github.com/ros2/ros2/wiki/Linux-Install-Debians)
+
+Setup your environment following [these instructions](https://github.com/ros2/ros2/wiki/Linux-Install-Debians#environment-setup)
+
+In the rest of these instruction we assume that every terminal setup the environment as instructed above.
+
+
+### Install from source
+
+You will need to have openssl installed on your machine:
+
+```bash
+sudo apt update && sudo apt install libssl-dev
+```
+
+First install ROS2 from source following [these instructions](https://github.com/ros2/ros2/wiki/Linux-Development-Setup)
+
+Note: Fast-RTPS requires an additional CMake flag to build the security plugins so the ament invocation needs to be modified to pass:
+```bash
+src/ament/ament_tools/scripts/ament.py build --build-tests --symlink-install --cmake-args -DSECURITY=ON --
+```
+
+### Additional configuration for RTI Connext
+
+Prerequisite: to use DDS-Scurity with connext you will need to procure an RTI Licence and install the security plugin.
+
+The RTI Connext installer allows you to choose where it lands in the filesystem.
+These instructions assume that you have prefixed the RTI paths with `$HOME/rti` so that the latest version (5.2.4 at time of writing) will land in `$HOME/rti/rti_connext_dds-5.2.4`.
+Note that the installer is a multi-partprocess.
+Fist you must install the "host" package, and then from its launcher you can install the additional "target" packages and the `secure_dds` package.
+Additional (and better) help is provided in the RTI documentation.
+
+```bash
+source ~/rti/rti_connext_dds-5.2.4/resource/scripts/rtisetenv_x64Linux3gcc4.8.2.bash'
+```
+
+It's often handy to create an alias for that super-long shell incantation to source the RTI script, for example, by adding something like this to your `~/.bashrc` file (altering the paths and alias names as needed/wanted):
+
+```
+alias rti='. ~/rti/rti_connext_dds-5.2.4/resource/scripts/rtisetenv_x64Linux3gcc4.8.2.bash'
+```
+
+## Run the demo
+
+### Create a folder for the files required by this demo
+
+We will now create a folder to store all the files necessary for this demo:
+
+```bash
+mkdir ~/sros2_demo
+```
+
+### Generating a keystore, keys and certificates
+
+#### Generate a keystore
+
+```bash
+cd ~/sros2_demo
+ros2 security create_keystore demo_keys
+```
+
+#### Generate keys and certificates for the talker and listener nodes
+
+```bash
+ros2 security create_key demo_keys talker
+ros2 security create_key demo_keys listener
+```
+
+### Define the SROS2 environment variables
+
+```bash
+export ROS_SECURITY_ROOT_DIRECTORY=$(pwd)/demo_keys
```

Comment:
I would suggest to use `~/sros2_demo` instead of `${pwd}` since the user might just change the cwd accidentally.